cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -661,58 +663,98 @@ impl TryFrom<PackageGroup> for api_types::PackageGroup {
-//         } else if let Some(msg) = value.subtype_1 {
-//             let inner = ResolutionMessageInner {
-//                 // FIXME: there's an error in the schema that turns this field into something other
-//                 //        than MessageLevel
-//                 level: MessageLevel::Error,
-//                 context: msg.context,
-//             };
-//             Ok(ResolutionMessage::AttrPathNotFound(inner))
-//         } else if let Some(msg) = value.subtype_2 {
-//             let inner = ResolutionMessageInner {
-//                 // FIXME: there's an error in the schema that turns this field into something other
-//                 //        than MessageLevel
-//                 level: MessageLevel::Error,
-//                 context: msg.context,
-//             };
-//             Ok(ResolutionMessage::ConstraintsTooTight(inner))
-//         } else {
-//             unreachable!("message was empty")
-//         }
-//     }
-// }
+/// The content of a generic message
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct MsgGeneral {
+    /// The log level of the message
+    pub level: Option<MessageLevel>,
+    /// The actual message
+    pub msg: String,
+}
+
+/// The content of a "attr path not found" message
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct MsgAttrPathNotFound {
+    /// The log level of the message
+    pub level: Option<MessageLevel>,
+    /// The actual message
+    pub msg: String,
+    /// The requested attribute path
+    pub attr_path: String,
+    /// The install id that requested this attribute path
+    pub install_id: String,
+    /// The systems on which this attribute path is valid
+    pub valid_systems: Vec<System>,
+}
+
+/// The content of a "constraints too tight" message
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct MsgConstraintsTooTight {
+    /// The log level of the message
+    pub level: Option<MessageLevel>,
+    /// The actual message
+    pub msg: String,
+}
+
+/// The kinds of resolution messages we can receive
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ResolutionMessage {
+    /// A generic message about resolution
+    General(MsgGeneral),
+    /// The attribute path requested for an install id either doesn't exist at all,
+    /// or isn't available on this system
+    AttrPathNotFound(MsgAttrPathNotFound),
+    /// Couldn't resolve a package group because the constraints were too tight,
+    /// which could mean that all the version constraints can't be satisfied by
+    /// a single page.
+    ConstraintsTooTight(MsgConstraintsTooTight),
+}
+
+impl ResolutionMessage {
+    pub fn msg(&self) -> String {
+        match self {
+            ResolutionMessage::General(msg) => msg.msg.clone(),
+            ResolutionMessage::AttrPathNotFound(msg) => msg.msg.clone(),
+            ResolutionMessage::ConstraintsTooTight(msg) => msg.msg.clone(),
+        }
+    }
+}
+
+impl From<MessagesItem> for ResolutionMessage {
```

Comment:
nonblocking: is the point of this pretty much just dropping context and type?

Reply:
And just generally having a layer of insulation between `flox-rust-sdk` and `catalog_api_v1::types`